feature/xhr/aggregate/index.js

```diff
@@ -104,62 +105,87 @@ baseEE.on('interactionDiscarded', function (interaction) {
 })
 
 function prepareHarvest(options) {
+  options = options || {}
+
   if (ajaxEvents.length === 0) {
     return null
   }
 
-  var payload = getPayload(ajaxEvents)
+  var payload = getPayload(ajaxEvents, options.maxPayloadSize || MAX_PAYLOAD_SIZE)
+
+  var payloadObjs = []
+  for (var i = 0; i < payload.length; i++) {
+    payloadObjs.push({body: {e: payload[i]}})
+  }
 
-  if (options && options.retry) {
+  if (options.retry) {
     sentAjaxEvents = ajaxEvents.slice()
   }
 
   ajaxEvents = []
 
-  return { body: { e: payload } }
+  return payloadObjs
 }
 
-function getPayload (events) {
-  var addString = getAddStringContext()
-  var payload = 'bel.7;'
-
-  for (var i = 0; i < events.length; i++) {
-    var event = events[i]
-
-    payload += '2,'
-
-    var fields = [
-      numeric(event.startTime),
-      numeric(event.endTime - event.startTime),
-      numeric(0), // callbackEnd
-      numeric(0), // no callbackDuration for non-SPA events
-      addString(event.method),
-      numeric(event.status),
-      addString(event.domain),
-      addString(event.path),
-      numeric(event.requestSize),
-      numeric(event.responseSize),
-      event.type === 'fetch' ? 1 : '',
-      addString(0), // nodeId
-      nullable(null, addString, true) + // guid
-      nullable(null, addString, true) + // traceId
-      nullable(null, numeric, false) // timestamp
-    ]
-
-    // add custom attributes
-    var attrParts = addCustomAttributes(loader.info.jsAttributes || {}, addString)
-    fields.unshift(numeric(attrParts.length))
-
-    payload += fields.join(',')
-
-    if (attrParts && attrParts.length > 0) {
-      payload += ';' + attrParts.join(';')
+function getPayload (events, maxPayloadSize, chunks) {
+  chunks = chunks || 1
+  var adders = []
+  var ver = 'bel.7;'
+  var payload = []
+  var chunkSize = events.length / chunks
+  var eventChunks = chunk(events, chunkSize)
+
+  for (var i = 0; i < eventChunks.length; i++) {
+    adders.push(getAddStringContext())
+    payload.push(ver)
+    var currentChunk = eventChunks[i]
+
+    for (var j = 0; j < currentChunk.length; j++) {
```

Comment:
This is a detail, but it might be easier to extract serialization of a single chunk into a separate function. It would make this more understandable - no need to have an array of adders and nested for loops.